conf/include/devupstream.inc

```diff
@@ -25,6 +25,11 @@
 BBCLASSEXTEND += "devupstream:target"
 
 python devupstream_virtclass_handler:prepend () {
+    deps = d.getVar("DEPENDS")
+    dev_deps = d.getVar("DEPENDS:class-devupstream")
+    if dev_deps:
+        d.setVar("DEPENDS", deps + dev_deps, parsing=True)
+
```

Comment:
Would it work if we replace this with:
```
d.appendVar("CLASSOVERRIDE", "class-devupstream")
```
I half suspect that [devupstream.bbclass](https://git.yoctoproject.org/poky/tree/meta/classes/devupstream.bbclass?h=kirkstone) erroneously adds an extra semicolon (`:`) to what it appends to `CLASSOVERRIDE`.
If this idea work, we might want to file a bug/propose a patch on poky.

Reply:
Actually the solution was easier than all of that:

``` diff
diff --git a/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb b/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb
index d6e3500..b1b25cd 100644
--- a/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb
+++ b/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb
@@ -32,3 +32,6 @@ PACKAGECONFIG[introspection] = "-DENABLE_INTROSPECTION=ON,-DENABLE_INTROSPECTION
 
 # Layer-Based SVG Engine
 PACKAGECONFIG[lbse] = "-DENABLE_LAYER_BASED_SVG_ENGINE=ON,-DENABLE_LAYER_BASED_SVG_ENGINE=OFF, "
+
+# unifdef-native: Needed since >2.38.
+DEPENDS:append:class-devupstream = " unifdef-native"
```
This results on this: 

```
# $DEPENDS [14 operations]
#   :append[pn-pixman:mips:toolchain-clang] /home/bot/yocto-rpi4-manual/sources/meta-clang/conf/nonclangable.conf:439
#     " openmp"
#   set /home/bot/yocto-rpi4-manual/sources/poky/meta/conf/bitbake.conf:305
#     ""
#   set /home/bot/yocto-rpi4-manual/sources/poky/meta/conf/documentation.conf:136
#     [doc] "Lists a recipe's build-time dependencies (i.e. other recipe files)."
#   :prepend /home/bot/yocto-rpi4-manual/sources/poky/meta/classes/base.bbclass:73
#     "${BASEDEPENDS} "
#   set /home/bot/yocto-rpi4-manual/sources/meta-webkit/recipes-browser/wpewebkit/wpewebkit.inc:14
#     "     ${@bb.utils.contains_any('LAYERSERIES_CORENAMES', 'dunfell gatesgarth hardknott honister', 'libsoup-2.4', 'libsoup', d)}     bison-native gperf-native harfbuzz-native libxml2-native ccache-native ninja-native ruby-native cairo     fontconfig freetype glib-2.0 harfbuzz icu jpeg pcre sqlite3 zlib libpng     libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt "
#   :prepend /home/bot/yocto-rpi4-manual/sources/poky/meta/classes/cmake.bbclass:4
#     "cmake-native "
#   :prepend /home/bot/yocto-rpi4-manual/sources/poky/meta/classes/pkgconfig.bbclass:1
#     "pkgconfig-native "
#   append /home/bot/yocto-rpi4-manual/sources/poky/meta/classes/perlnative.bbclass:2
#     "perl-native"
#   :append /home/bot/yocto-rpi4-manual/sources/poky/meta/classes/python3native.bbclass:5
#     " python3-native "
#   append /home/bot/yocto-rpi4-manual/sources/meta-webkit/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb:13
#     " libwpe"
>>>>>> #   :append[class-devupstream] /home/bot/yocto-rpi4-manual/sources/meta-webkit/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb:37
#     " unifdef-native"
#   append base.bbclass:490 [appendVar]
#     " libavif libjxl gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-good openjpeg gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good libgcrypt libtasn1 woff2"
#   append cmake.bbclass:19 [__anon_24__home_bot_yocto_rpi4_manual_sources_poky_meta_classes_cmake_bbclass]
#     " ninja-native"
#   override[pn-linux-firmware]::append[toolchain-clang] /home/bot/yocto-rpi4-manual/sources/meta-clang/conf/nonclangable.conf:436
#     " clang-native"
# pre-expansion value:
#   "pkgconfig-native cmake-native ${BASEDEPENDS}      ${@bb.utils.contains_any('LAYERSERIES_CORENAMES', 'dunfell gatesgarth hardknott honister', 'libsoup-2.4', 'libsoup', d)}     bison-native gperf-native harfbuzz-native libxml2-native ccache-native ninja-native ruby-native cairo     fontconfig freetype glib-2.0 harfbuzz icu jpeg pcre sqlite3 zlib libpng     libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt  perl-native  libwpe python3-native  unifdef-native libavif libjxl gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-good openjpeg gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good libgcrypt libtasn1 woff2 ninja-native"
DEPENDS="pkgconfig-native cmake-native virtual/aarch64-poky-linux-gcc virtual/aarch64-poky-linux-compilerlibs virtual/libc      libsoup     bison-native gperf-native harfbuzz-native libxml2-native ccache-native ninja-native ruby-native cairo     fontconfig freetype glib-2.0 harfbuzz icu jpeg pcre sqlite3 zlib libpng     libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt  perl-native  libwpe python3-native  unifdef-native libavif libjxl gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-good openjpeg gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good libgcrypt libtasn1 woff2 ninja-native"
#
```

`DEPENDS="pkgconfig-native  ... unifdef-native  ..."

Reply:
This works for kirkstone (presumably for older releases too).